src/RetryMiddleware.php

```diff
@@ -2,11 +2,8 @@
 namespace Aws;
 
 use Aws\Exception\AwsException;
-use Exception;
 use Psr\Http\Message\RequestInterface;
-use GuzzleHttp\Promise;
```

Comment:
It looks like this was intended to alleviate the full namespace reference on [line 136](https://github.com/aws/aws-sdk-php/pull/1453/files#diff-85d927b5b18f9813bfd8535665c1e98dR136).

This could be `return Promise\rejection_for(` instead.